### PR TITLE
Add buttons to search alerts & LC archive near source on source page

### PR DESCRIPTION
**Pull request by acrellin**
This PR introduces a button to the source page that, when clicked, submits a request to query alerts within 3 arcseconds of the source in question and redirects to the alerts page to render the results. Other improvements to the alerts page, including a spinner during request processing, and redux state structure updates, are included.

Closes  https://github.com/fritz-marshal/fritz/issues/287

alerts_search_button

### CODE REVIEW

**Comment by acrellin** [comment]
CI failure unrelated

File: extensions/skyportal/static/js/components/ArchiveSearchButton.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import React, { useEffect } from "react";
+import PropTypes from "prop-types";
+import { useDispatch, useSelector } from "react-redux";
+import { Link } from "react-router-dom";
+import Button from "@material-ui/core/Button";
+
+import { showNotification } from "baselayer/components/Notifications";
+import * as archiveActions from "../ducks/archive";
+
+const AlertsSearchButton = ({ ra, dec, radius = 3 }) => {
```
Comment (kmshin1397) [inline extensions/skyportal/static/js/components/ArchiveSearchButton.jsx:10]:
```suggestion
const ArchiveSearchButton = ({ ra, dec, radius = 3 }) => {
```

File: extensions/skyportal/static/js/components/ArchiveSearchButton.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  };
+
+  return (
+    <Link to="/archive" onClick={handleClick}>
+      <Button variant="contained">View ZTF Light Curve Archive</Button>
+    </Link>
+  );
+};
+
+AlertsSearchButton.propTypes = {
```
Comment (kmshin1397) [inline extensions/skyportal/static/js/components/ArchiveSearchButton.jsx:51]:
```suggestion
ArchiveSearchButton.propTypes = {
```

File: extensions/skyportal/static/js/components/ArchiveSearchButton.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  objID: PropTypes.string.isRequired,
+  ra: PropTypes.number.isRequired,
+  dec: PropTypes.number.isRequired,
+  radius: PropTypes.number,
+};
+AlertsSearchButton.defaultProps = {
```
Comment (kmshin1397) [inline extensions/skyportal/static/js/components/ArchiveSearchButton.jsx:57]:
```suggestion
ArchiveSearchButton.defaultProps = {
```

File: extensions/skyportal/static/js/components/ArchiveSearchButton.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  radius: 3,
+};
+
+export default AlertsSearchButton;
```
Comment (kmshin1397) [inline extensions/skyportal/static/js/components/ArchiveSearchButton.jsx:61]:
```suggestion
export default ArchiveSearchButton;
```

**Review by kmshin1397 — CHANGES_REQUESTED** [review]
Looks good other than some renaming, thanks @acrellin!

**Comment by acrellin** [comment]
@kmshin1397 ah right, of course. Fixed!

**Review by kmshin1397 — APPROVED** [review]

File: extensions/skyportal/static/js/components/ArchiveSearchButton.jsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+          "Catalog names could not be fetched; enter search criteria manually",
+          "warning"
+        )
+      );
+    }
+  };
+
+  return (
+    <Link to="/archive" onClick={handleClick}>
+      <Button variant="contained">View ZTF Light Curve Archive</Button>
```
Comment (dmitryduev) [inline extensions/skyportal/static/js/components/ArchiveSearchButton.jsx:46]:
```suggestion
      <Button variant="contained">Search ZTF Light Curve Archive</Button>
```

File: extensions/skyportal/static/js/components/AlertsSearchButton.jsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+const AlertsSearchButton = ({ objID, ra, dec, radius = 3 }) => {
+  const dispatch = useDispatch();
+  const handleClick = () => {
+    dispatch(alertsActions.fetchAlerts({ object_id: objID, ra, dec, radius }));
+  };
+
+  return (
+    <Link to="/alerts" onClick={handleClick}>
+      <Button variant="contained">
+        Search Nearby Alerts
```
Comment (dmitryduev) [inline extensions/skyportal/static/js/components/AlertsSearchButton.jsx:18]:
```suggestion
        Search ZTF Alert Archive
```

File: extensions/skyportal/static/js/components/AlertsSearchButton.jsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import { useDispatch } from "react-redux";
+import { Link } from "react-router-dom";
+import Button from "@material-ui/core/Button";
+
+import * as alertsActions from "../ducks/alerts";
+
+const AlertsSearchButton = ({ objID, ra, dec, radius = 3 }) => {
+  const dispatch = useDispatch();
+  const handleClick = () => {
+    dispatch(alertsActions.fetchAlerts({ object_id: objID, ra, dec, radius }));
```
Comment (dmitryduev) [inline extensions/skyportal/static/js/components/AlertsSearchButton.jsx:12]: There may be sources with other `objectId`'s within 3 arcseconds - let's not restrict the search to the current `objID` only:

```suggestion
    dispatch(alertsActions.fetchAlerts({ ra, dec, radius }));
```

File: extensions/skyportal/static/js/components/SourceDesktop.jsx [hunk 8]
```diff
@@ -244,6 +246,16 @@ const SourceDesktop = ({ source }) => {
               <TNSInfo objID={source.id} />
             </div>
           </div>
+          <div className={classes.infoLine}>
+            <div className={classes.sourceInfo}>
+              <AlertsSearchButton objID={source.id} ra={source.ra} dec={source.dec} />
```
Comment (dmitryduev) [inline extensions/skyportal/static/js/components/SourceDesktop.jsx:251]:
```suggestion
              <AlertsSearchButton ra={source.ra} dec={source.dec} />
```

File: extensions/skyportal/static/js/components/AlertsSearchButton.jsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { useDispatch } from "react-redux";
+import { Link } from "react-router-dom";
+import Button from "@material-ui/core/Button";
+
+import * as alertsActions from "../ducks/alerts";
+
+const AlertsSearchButton = ({ objID, ra, dec, radius = 3 }) => {
```
Comment (dmitryduev) [inline extensions/skyportal/static/js/components/AlertsSearchButton.jsx:9]:
```suggestion
const AlertsSearchButton = ({ ra, dec, radius = 3 }) => {
```

File: extensions/skyportal/static/js/components/AlertsSearchButton.jsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    <Link to="/alerts" onClick={handleClick}>
+      <Button variant="contained">
+        Search Nearby Alerts
+      </Button>
+    </Link>
+  );
+};
+
+AlertsSearchButton.propTypes = {
+  objID: PropTypes.string.isRequired,
```
Comment (dmitryduev) [inline extensions/skyportal/static/js/components/AlertsSearchButton.jsx:25]:
```suggestion
```

**Review by dmitryduev — CHANGES_REQUESTED** [review]
Looks great, thanks @acrellin! A few requests:

Could you please: 
- Make the buttons the same size (i.e. smaller) as the ones below? 
- Move them up so that they appear right after the coordinates (since we'll be querying by coordinates)?
- Unrelated to this PR, but I'd swap redshift and TNS match info in the header - the latter is also done by coordinates, so would be more natural to appear after the buttons from this PR.
- How hard would it be to fill out the forms on the Alerts/Archive pages with the ra/dec/radius info so that the user can see what was searched?

**Comment by acrellin** [comment]
@dmitryduev I've addressed all of your comments except for the form population which would require storing the form-values-to-be in redux state and populating the component's inputs accordingly. Doesn't seem like there's much added benefit for that work (they can easily see the coords of the source from which they clicked search). Thoughts?

**Review by dmitryduev — APPROVED** [review]
👍🏻 
re: autofill - let's leave it for now.

**Pull request merged**